### DISCUSSION TITLE
Recommended font drawing crate?

### DISCUSSION THREAD

**Discussion by junglie85** [parasyte/pixels#359] [General]
Are there any font drawing crates that are known to work well with pixels?

**Comment by parasyte** [comment]
I haven't used any font crates with it. In theory, anything that can rasterize a font would work.

[`fontdue`](https://crates.io/crates/fontdue) gives out `Vec<u8>` for each character you rasterize. (It doesn't return images, it's a coverage grid. But it can be converted into an image.) Combined with a crate like [`pix`](https://crates.io/crates/pix), should give you a decent headstart.

**Reply by junglie85** [reply]
That’s a great idea. Thanks.
